### PR TITLE
feat: implement all compliance checks

### PR DESCRIPTION
**Pull request by SgtPooki**
Reports are generating properly locally, but consumption and custom runs have not been tested nor documented thoroughly.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,60 @@
 # pinning-service-compliance
+
+> The compliance test suite for [IPFS Pinning Service API Spec](https://ipfs.github.io/pinning-services-api-spec/)
+
+## Getting started
+
```
Comment (lidel) [inline README.md:6]: The first thing in README should be the one-liner that runs this from NPM.

We can't expect service providers to build this by hand. They will use a prebuilt version from NPM.  Filled #22

File: src/output/getReportEntry.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+##### Headers
+\`\`\`json
+${stringifyHeaders(request.headers)}
+\`\`\`
+##### Body
+\`\`\`json
+${request.body}
+\`\`\`
+#### Response data from ${url}
+*via util.inspect*
```
Comment (lidel) [inline src/output/getReportEntry.ts:49]:
```suggestion
```

File: src/output/getReportEntry.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+\`\`\`
+${inspect(response.json, { depth: 4 })}
+\`\`\`
+#### Response data after being parsed by RemotePinningServiceClient
+*via util.inspect*
```
Comment (lidel) [inline src/output/getReportEntry.ts:54]:
```suggestion
```

**Review by lidel — CHANGES_REQUESTED** [review]
I did a quick pass, and aside from small papercuts (see comments inline), looks good.

Ping me when these are addressed, or if you need help with CI/semantic release setup.

File: .vscode/launch-copy.json [hunk 4]
```diff
@@ -0,0 +1,21 @@
+{
```
[the remark this answers is not in the document]
Comment (SgtPooki) [reply .vscode/launch-copy.json:1]: done.

**Pull request merged**